great_expectations/validator/validator.py

```diff
@@ -360,6 +375,298 @@ def inst_expectation(*args, **kwargs):
+
+        override_variables: Optional[Dict[str, Any]] = override_profiler_config.get(
+            "variables"
+        )
+        if override_variables is None:
+            override_variables = {}
+
+        effective_variables: Optional[
+            ParameterContainer
+        ] = profiler.reconcile_profiler_variables(variables=override_variables)
+        profiler.variables = effective_variables
+
+        override_rules: Optional[
+            Dict[str, Dict[str, Any]]
+        ] = override_profiler_config.get("rules")
+        if override_rules is None:
+            override_rules = {}
+
+        assert (
+            len(override_rules) <= 1
+        ), "An override Rule-Based Profiler for an Expectation can have exactly one rule."
+
+        if override_rules:
+            profiler.rules[0].name = list(override_rules.keys())[0]
+
+            effective_rules: List[Rule] = profiler.reconcile_profiler_rules(
+                rules=override_rules,
+                reconciliation_directives=ReconciliationDirectives(
+                    domain_builder=ReconciliationStrategy.UPDATE,
+                    parameter_builder=ReconciliationStrategy.REPLACE,
+                    expectation_configuration_builder=ReconciliationStrategy.REPLACE,
+                ),
+            )
+            profiler.rules = effective_rules
+
+            rule = profiler.rules[0]
+
+            self._validate_rule_and_update_rule_properties(
+                rule=rule,
+                expectation_type=expectation_type,
+                expectation_kwargs=expectation_kwargs,
+                success_keys=success_keys,
+            )
+
+        return profiler
+
+    def _validate_rule_and_update_rule_properties(
+        self,
+        rule: Rule,
+        expectation_type: str,
+        expectation_kwargs: dict,
+        success_keys: Tuple[str],
+        ignore_batch_request: bool = False,
+    ) -> None:
+        assert (
+            rule.expectation_configuration_builders[0].expectation_type
+            == expectation_type
+        ), "ExpectationConfigurationBuilder in profiler used to build an ExpectationConfiguration must have the same expectation_type as the expectation being invoked."
+
+        domain_type: MetricDomainTypes = rule.domain_builder.domain_type
+        # TODO: <Alex>Handle future domain_type cases as they are defined.</Alex>
+        if domain_type == MetricDomainTypes.COLUMN:
+            column_name = expectation_kwargs["column"]
+            rule.domain_builder.column_names = [column_name]
+            if ignore_batch_request or rule.domain_builder.batch_request is None:
+                # A DomainBuilder that emits MetricDomainTypes.COLUMN type Domain object needs exactly 1 Batch of data.
+                rule.domain_builder.batch = self.active_batch
+        elif domain_type == MetricDomainTypes.TABLE:
+            pass  # No action is needed.
+        else:
+            pass  # No action is needed.
+
```

Comment:
Any logging?

Reply:
@cdkini Logging is not needed here, because it is open to new domains to be needed; until then, no action.